File: Four/License.php
<?php
abstract class License {
    private string $type;
    private float $cost;
    private int $duration; // In years

    public function __construct(string $type, float $cost, int $duration) {
        $this->type = $type;
        $this->cost = $cost;
        $this->duration = $duration;
    }

    public function getType(): string {
        return $this->type;
    }

    public function getCost(): float {
        return $this->cost;
    }

    public function getDuration(): int {
        return $this->duration;
    }

    abstract public function calculateTotalCost(): float;
}
?>

File: java/Four/Four/License.java
public abstract class License {
    protected String type;
    protected double cost;
    protected int duration; // In years

    public License(String type, double cost, int duration) {
        this.type = type;
        this.cost = cost;
        this.duration = duration;
    }

    public String getType() {
        return type;
    }

    public double getCost() {
        return cost;
    }

    public int getDuration() {
        return duration;
    }

    public abstract double calculateTotalCost();
}